pkg/sfu/receiver.go

```diff
@@ -568,17 +569,20 @@ func (w *WebRTCReceiver) forwardRTP(layer int32) {
 		}
 
 		w.downTrackMu.RLock()
-		if w.lbThreshold == 0 || len(w.downTracks)-len(w.free) < w.lbThreshold {
+		downTracks := w.downTracks
```

Comment:
Nice!!!

What is the 'free' variable here?

Also in Golang , is the object still valid as long this reference is held? Trying to understand what happens if a down track unsubscribes while this copied loop is running?

Reply:
Also, is this a regression due to the sequence number/VP8 picture id minters I introduced. Guessing this was working before without issues.

Reply:
> What is the 'free' variable here?

free keeps track of indices that are unused. This is used when we remove downtracks, so we could avoid copying the slice.

> Also in Golang , is the object still valid as long this reference is held? Trying to understand what happens if a down track unsubscribes while this copied loop is running?

yes, it'll grab a copy of the slice (start position, length, capacity). If unsubscribed during a read cycle, it'll either retain the last DownTrack, or nil. 

> Also, is this a regression due to the sequence number/VP8 picture id minters I introduced. Guessing this was working before without issues.

It doesn't appear to be related. We were able to reproduce it on versions before that was introduced.

Reply:
Thank you David.